### PR TITLE
[scripts-audit] gn buildsystem

### PR DESCRIPTION
**Pull request by strega-nil**
See issue #16188

This removes `vcpkg_gn_build(`, since it's never been used and is just a pass-through to `vcpkg_build_ninja(` (needs discussion, may end up being reverted). Additionally, `vcpkg_gn_configure` and `vcpkg_gn_install` are audited.

### CODE REVIEW

**Comment by Neumann-A** [comment]
`vcpkg_gn_configure` probably needs get the compiler flags from the `CMAKE_TOOLCHAIN` to be consistent.

**Comment by strega-nil-ms** [comment]
@Neumann-A these scripts audits are not modifying the existing scripts at all; they don't add functionality, in order to allow me to actually get through all the scripts.

File: scripts/cmake/vcpkg_install_gn.cmake [hunk 1]
```diff
@@ -21,80 +21,116 @@ Only install the specified targets.
...
+    endif()
 
-    function(gn_get_target_type OUT_VAR BUILD_DIR TARGET)
-        execute_process(
-            COMMAND ${GN} desc "${BUILD_DIR}" "${TARGET}"
-            WORKING_DIRECTORY "${_vig_SOURCE_PATH}"
-            OUTPUT_VARIABLE OUTPUT_
-            OUTPUT_STRIP_TRAILING_WHITESPACE
+    execute_process(
+        COMMAND ${GN} desc "${arg_BUILD_DIR}" "${arg_TARGET}" "${arg_WHAT_TO_DISPLAY}"
```
Comment (strega-nil) [inline scripts/cmake/vcpkg_install_gn.cmake:50]:
```suggestion
        COMMAND "${GN}" desc "${arg_BUILD_DIR}" "${arg_TARGET}" "${arg_WHAT_TO_DISPLAY}"
```

**Review by strega-nil — COMMENTED** [review]
Note: add docs for `vcpkg_build_ninja`

**Pull request merged**